### PR TITLE
Initial recipe for gis-metadata-parser

### PR DESCRIPTION
**Pull request by jlaura**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by jlaura** [comment]
@conda-forge/help-python This should be all good to go. One more package inbound after this and then the full dependency stack will be available on Conda-forge. Thanks for all the hard work!

File: recipes/gis-metadata-parser/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+requirements:
+  host:
+    - pip
+    - poetry
+    - python >=3.6,<4.0
+  run:
+    - frozendict >=2.0,<3.0
+    - parserutils >=2.0.1,<3.0.0
+    - python >=3.6,<4.0
```
Comment (dopplershift) [inline recipes/gis-metadata-parser/meta.yaml:26]: Is there some reason you're adding upper bounds that [upstream doesn't have](https://github.com/consbio/gis-metadata-parser)?

Comment (jlaura) [reply recipes/gis-metadata-parser/meta.yaml:26]: Only that these bounds  came direct from [castle grayskull](https://github.com/conda-incubator/grayskull). I am more than happy to pull them.

Comment (jlaura) [reply recipes/gis-metadata-parser/meta.yaml:26]: I think that these are also actually being parsed out of the [poetry.lock file](https://github.com/consbio/gis-metadata-parser/blob/main/poetry.lock#L30) by gray skull. For example, the parser-utils dep there specifies python >=3.6 <4.0. I think grayskull is correct in how it specified these deps based on the lock file.

Comment (dopplershift) [reply recipes/gis-metadata-parser/meta.yaml:26]: Ah, they came from translating the `~=` in `pyproject.toml`. Fair enough.

**Review by dopplershift — APPROVED** [review]

**Pull request merged**